scripts/approve-fingerprint-sudo.sh

```diff
@@ -5,10 +5,19 @@
 # To be able to use the mac touchbar for authorizing the 'sudo' command in terminal
 # This will persist through software updates unlike changes directly made to '/etc/pam.d/sudo'
 # Copied from: https://apple.stackexchange.com/a/466029
-# TODO: Need to ensure that TouchId hardware is present before running this script
 
 type is_file &> /dev/null 2>&1 || source "${HOME}/.shellrc"
 
+function check_touchid_hardware() {
+  ioreg -c AppleBiometricSensor | grep -q "AppleBiometricSensor"
+  return $? # Exit status 0 if found, non-zero otherwise
+}
+
+if ! check_touchid_hardware; then
+  echo "Touch ID hardware is not detected. Exiting script."
+  exit 1
```

Comment:
Since this script will be called from the `fresh-install` script, it should not return/exit with a non-zero exit code. The rest of the installation/backup process should continue. So, please remove this line